sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/external/JdbcExternalWrite.java

```diff
@@ -0,0 +1,155 @@
+
+/** Exposes {@link JdbcIO.Write} as an external transform for cross-language usage. */
+@Experimental(Kind.PORTABILITY)
+@AutoService(ExternalTransformRegistrar.class)
+public class JdbcExternalWrite implements ExternalTransformRegistrar {
+
+  public static final String URN = "beam:external:java:jdbc:write:v1";
+
+  @Override
+  public Map<String, Class<? extends ExternalTransformBuilder>> knownBuilders() {
+    return ImmutableMap.of(URN, JdbcExternalWrite.Builder.class);
+  }
+
+  /** Parameters class to expose the Write transform to an external SDK. */
+  public static class Configuration {
+    private String statement;
+    private String driverClassName;
+    private String jdbcUrl;
+    private String username;
+    private String password;
+    private String connectionProperties;
+    private Iterable<String> connectionInitSqls;
+
+    public void setStatement(String statement) {
+      this.statement = statement;
+    }
+
+    public void setDriverClassName(String driverClassName) {
+      this.driverClassName = driverClassName;
+    }
+
+    public void setJdbcUrl(String jdbcUrl) {
+      this.jdbcUrl = jdbcUrl;
+    }
+
+    public void setUsername(String username) {
+      this.username = username;
+    }
+
+    public void setPassword(String password) {
+      this.password = password;
+    }
+
+    public void setConnectionProperties(String connectionProperties) {
+      this.connectionProperties = connectionProperties;
+    }
+
+    public void setConnectionInitSqls(Iterable<String> connectionInitSqls) {
+      this.connectionInitSqls = connectionInitSqls;
+    }
+  }
+
+  public static class Builder
+      implements ExternalTransformBuilder<
+          JdbcExternalWrite.Configuration, PCollection<Row>, PDone> {
+    @Override
+    public PTransform<PCollection<Row>, PDone> buildExternal(
+        JdbcExternalWrite.Configuration configuration) {
+      DataSourceConfiguration dataSourceConfiguration =
+          DataSourceConfiguration.create(configuration.driverClassName, configuration.jdbcUrl)
+              .withUsername(configuration.username)
+              .withPassword(configuration.password)
+              .withConnectionProperties(configuration.connectionProperties);
+
+      if (configuration.connectionInitSqls != null) {
+        List<String> connectionInitSqls =
+            StreamSupport.stream(configuration.connectionInitSqls.spliterator(), false)
+                .collect(Collectors.toList());
+        dataSourceConfiguration =
+            dataSourceConfiguration.withConnectionInitSqls(connectionInitSqls);
+      }
+
+      JdbcIO.PreparedStatementSetter<Row> preparedStatementSetter = new PrepareStatementXlangRow();
+
+      return JdbcIO.<Row>write()
+          .withDataSourceConfiguration(dataSourceConfiguration)
+          .withStatement(configuration.statement)
+          .withPreparedStatementSetter(preparedStatementSetter);
```

Comment:
It would be nice at some point to make this into a `JdbcIO.WriteRows` companion to `JdbcIO.ReadRows`, but doesn't need to happen as part of this PR.

Reply:
When I was writing Read xlang transform I noticed there are 2 of them and in read I use ReadRows. But I agree it could be done in another task.